python/src/py_stream/mod.rs

```diff
@@ -36,6 +39,94 @@ impl PyStream {
     fn id(&self) -> String {
         format!("{}", self.id)
     }
+
+    fn _map(&self, py: Python<'_>, function: PyObject) -> PyResult<Py<PyOperatorStream>> {
```

Comment:
```suggestion
    fn map(&self, py: Python<'_>, function: PyObject) -> PyResult<Py<PyOperatorStream>> {
```

`PyStream` is an internal API already, so we don't need to indicate its methods as internal.

Reply:
I used `_map` because otherwise there is a name clash from the `map` method being implemented atop the `Stream` trait in Rust.

When creating a new `PyOperatorStream` below, the method then defaults to this `map` implementation instead of the trait one.